sklearn/tests/test_naive_bayes.py

```diff
@@ -647,15 +616,6 @@ def test_cnb():
     assert_array_almost_equal(clf.feature_log_prob_, normed_weights)
 
 
-def test_naive_bayes_scale_invariance():
-    # Scaling the data should not change the prediction results
-    iris = load_iris()
-    X, y = iris.data, iris.target
-    labels = [GaussianNB().fit(f * X, y).predict(f * X)
-              for f in [1E-10, 1, 1E10]]
-    assert_array_equal(labels[0], labels[1])
-    assert_array_equal(labels[1], labels[2])
```

Comment:
Why moving this test around in the file? It makes the diff hard to follow.

Reply:
Thanks for your remark. I didn't expect go get feedback so quickly. (I commented in a review many of my changes).

Basically, I moved tests around to order them by the class that is tested. Right now (in master), it is a bit mixed, which makes it unnecessarily complicated to get an overview of what is tested and what not. E.g. gnb tests come in between discrete nb tests and tests are not always named in the same pattern, so by just reading the function name, it is not obvious what is tested.
Unfortunately, I saw no way to refactor the tests without moving them.

Reply:
> [..] 
> This will also make using git blame more difficult without good reason.
> 
> I would suggest you look at the diff on Github and try to reduce it somewhat, at least by not moving functions around.

Moving the functions was a central idea of this PR. Do you think it is more important to have a good git blame history or have the codebase well organized?

Reply:
git blame history is more useful for maintenance.